### PR TITLE
feat(thorvg): update thorvg to 0.15.3

### PR DESCRIPTION
**Pull request by lhdjply**
A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by lhdjply** [comment]
cc @kisvegabor  The incorrect 32-bit images cannot be downloaded all at once; only a few can be downloaded each time, which is quite troublesome.

**Comment by lhdjply** [comment]
how to download the 32bit png @kisvegabor

**Comment by lhdjply** [comment]
cc @kisvegabor @FASTSHIFT @XuNeo @liamHowatt

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — COMMENTED** [review]
I'm ok with using the latest version, but I winder if there is a particular feature why ThorVG shall be updated?

These many changed files are too much for GitHub so I just sampled a few ref. images and they looked good. :slightly_smiling_face:

**Comment by lhdjply** [comment]
> I'm ok with using the latest version, but I winder if there is a particular feature why ThorVG shall be updated?
> 
> These many changed files are too much for GitHub so I just sampled a few ref. images and they looked good. 🙂

Because compared to the previous version, performance has been enhanced. https://github.com/thorvg/thorvg/issues/2747

image

**Review by XuNeo — CHANGES_REQUESTED** [review]
The shape size is changed, like this one: tests/ref_imgs/draw/svg_draw_svg_shapes_9.lp32.png

Some of the reference image's change is not visible. Could you confirm if it's actually changed? @FASTSHIFT Please help to confirm on your side too.

**Comment by kisvegabor** [comment]
So it's 10% performance gain for 20% extra memory usage which is not that appealing at first look. Anyway, sooner or later we need to update ThorVG so it's ok.

**Comment by lhdjply** [comment]
> The shape size is changed, like this one: tests/ref_imgs/draw/svg_draw_svg_shapes_9.lp32.png
> 
> Some of the reference image's change is not visible. Could you confirm if it's actually changed? @FASTSHIFT Please help to confirm on your side too.

image

All the generated images look the same as the original, but for some reason, they are judged to be different.

**Comment by lhdjply** [comment]
@XuNeo @kisvegabor  @FASTSHIFT  To facilitate the review, I have changed it to multiple commits. Now the images should be visible in each commit.

**Comment by kisvegabor** [comment]
If you run the tests when they fail they probably will show that there is a difference like `0xff3456` instead of `0xff3457` on an antialiased edge.

**Comment by XuNeo** [comment]
> The shape size is changed, like this one: tests/ref_imgs/draw/svg_draw_svg_shapes_9.lp32.png

The SVG shape size is changed. The new one is slightly larger. Maybe we should feedback to ThorVG.

image

**Comment by kisvegabor** [comment]
Reported here: https://github.com/thorvg/thorvg/issues/2908

**Comment by lhdjply** [comment]
<svg width="8cm" height="4cm" viewBox="0 0 800 400">
<g><defs><linearGradient id="MyGradient" gradientUnits="userSpaceOnUse"
 x1="0" y1="0" x2="350" y2="350">
<stop offset="0" stop-color="red"/>
<stop offset="0.5" stop-color="blue"/>
</linearGradient></defs>
<rect fill="url(#MyGradient)" stroke="black" stroke-width="5"
x="0" y="0" width="200" height="600"/></g></svg>

**Comment by lhdjply** [comment]
@XuNeo  thorvg 0.15.3 released, svg_draw_svg_shapes_9.lp32.png now is correct.

File: tests/ref_imgs_vg_lite/draw/blend_to_rgb565.png
Comment (XuNeo) [inline tests/ref_imgs_vg_lite/draw/blend_to_rgb565.png:1]: This test is broken before this update. There's no visible changes in the new png file. Could you confirm all of them what exactly is changed and why?

**Review by XuNeo — COMMENTED** [review]
Lots of images are updated, but no visible change. Could you please confirm what exactly changed? Thank you!

File: tests/ref_imgs_vg_lite/draw/blend_to_rgb565.png
Comment (lhdjply) [reply tests/ref_imgs_vg_lite/draw/blend_to_rgb565.png:1]: cc @hermet Can you help to look this pr ? Why so many png file was change? But it looks like nothing has changed.

Comment (hermet) [reply tests/ref_imgs_vg_lite/draw/blend_to_rgb565.png:1]: @lhdjply Hi, I think the rounding method for RGB pixels during blending has been modified for improving speed, with no noticeable visual difference.

Comment (lhdjply) [reply tests/ref_imgs_vg_lite/draw/blend_to_rgb565.png:1]: >  I think the rounding method for RGB pixels during blending has been modified for improving speed, with no noticeable visual difference.

@hermet Thank you!

**Comment by FASTSHIFT** [comment]
I used AI to help write a script for batch image comparison, and found that most of the differences were in the processing of anti-aliased edges, and no major problems were found.

```python
from PIL import Image, ImageChops
import os

# Set directory paths
dir1 = "lvgl_copy/tests/ref_imgs"
dir2 = "lvgl/tests/ref_imgs"
output_dir = "diff_sw"

# Ensure the output directory exists
if not os.path.exists(output_dir):
    os.makedirs(output_dir)

# Traverse all files in the first directory and its subdirectories
for root, dirs, files in os.walk(dir1):
    for file1 in files:
        if file1.endswith(".jpg") or file1.endswith(".png"):
            # Construct full path
            path1 = os.path.join(root, file1)
            # Calculate the corresponding file path in the second directory
            relative_path = os.path.relpath(path1, dir1)  # Calculate relative path
            path2 = os.path.join(dir2, relative_path)  # Combine to get full path

            # Check if the corresponding file exists in the second directory
            if os.path.exists(path2):
                # Open both images
                im1 = Image.open(path1)
                im2 = Image.open(path2)

                # Compare the two images
                diff = ImageChops.difference(im1, im2)

                # Generate a highlighted image of inconsistent pixels
                if diff.getbbox():  # If there are different parts
                    # Create a red overlay layer
                    red_overlay = Image.new("RGBA", im1.size, (255, 0, 0, 128))  # Red semi-transparent layer

                    # Create a mask with 255 only on the difference part
                    diff_mask = diff.convert("L")  # Convert the difference image to grayscale
                    diff_mask = diff_mask.point(lambda p: p > 0 and 255)  # Create a binary mask

                    # Apply the red highlight layer to the original image
                    combined = Image.composite(red_overlay, im1, diff_mask)

                    # Save the difference image
                    relative_output_path = os.path.relpath(path1, dir1)  # Get relative path
                    diff_output_path = os.path.join(output_dir, relative_output_path)
                    diff_output_dir = os.path.dirname(diff_output_path)

                    # Ensure the output directory exists
                    if not os.path.exists(diff_output_dir):
                        os.makedirs(diff_output_dir)

                    combined.save(diff_output_path)
                    print(f"Saved difference image: {diff_output_path}")
                else:
                    print(f"No difference: {path1}")
            else:
                print(f"File not found: {path2}")

```
[diff_imgs.zip](https://github.com/user-attachments/files/17631020/diff_imgs.zip)

**Review by FASTSHIFT — APPROVED** [review]

**Comment by kisvegabor** [comment]
Fantastic!

**Review by kisvegabor — APPROVED** [review]

**Comment by XuNeo** [comment]
> I used AI to help write a script for batch image comparison, and found that most of the differences were in the processing of anti-aliased edges, and no major problems were found.

That's all I want to confirm. Thank you.

**Comment by lhdjply** [comment]
> I used AI to help write a script for batch image comparison, and found that most of the differences were in the processing of anti-aliased edges, and no major problems were found.

Thank you!

**Review by XuNeo — APPROVED** [review]

**Pull request merged**